pandas/core/groupby.py

```diff
@@ -1487,27 +1489,40 @@ def aggregate(self, values, how, axis=0):
                 raise NotImplementedError
             out_shape = (self.ngroups,) + values.shape[1:]
 
-        if is_numeric_dtype(values.dtype):
-            values = com.ensure_float(values)
-            is_numeric = True
-            out_dtype = 'f%d' % values.dtype.itemsize
-        else:
-            is_numeric = issubclass(values.dtype.type, (np.datetime64,
-                                                        np.timedelta64))
+        if is_bool_dtype(values.dtype):
+            values = _algos.ensure_float64(values)
+        elif (com.is_integer_dtype(values) or
+                _is_datetime_or_timedelta_dtype(values.dtype)):
+            values = values.astype('int64')
```

Comment:
actually, I think this needs to go back to several if/else.

if its a datetime/timedeltas, then you do ``.view('int64')``
elif boolean: convert to float
elseif its an integer, then you can do: ``.astype('int64',copy=False)`` (or ``np.asarray(values,np.int64)``), which ii think is the same
else : convert to float